sn_node/src/bin/safenode/main.rs

```diff
@@ -296,6 +296,25 @@ fn monitor_node_events(mut node_events_rx: NodeEventsReceiver, ctrl_tx: mpsc::Se
     let _handle = tokio::spawn(async move {
         loop {
             match node_events_rx.recv().await {
+                Ok(NodeEvent::AttemptingNetworkConnection) => {
+                    Marker::AttemptingNetworkConnection.log();
+                    trace!("Attempting to connect to the network: dialing initial peers");
+                    let inactivity_timeout = Instant::now();
+
+                    loop {
+                        if inactivity_timeout.elapsed() >= Duration::from_secs(60) {
```

Comment:
This will block the event receiver for 60 seconds. That means other events will not be handled for this period.